pkg/roachprod/roachprod.go

```diff
@@ -751,7 +761,34 @@ func Start(
 	if err != nil {
 		return err
 	}
-	return c.Start(ctx, l, startOpts)
+	if err = c.Start(ctx, l, startOpts); err != nil {
+		return err
+	}
+	createClusterConfig(ctx, l, c, startOpts)
+	return nil
+}
+
+// createClusterConfig creates the prometheus instance cluster config. Any error is logged and ignored.
+func createClusterConfig(ctx context.Context, l *logger.Logger, c *install.SyncedCluster, startOpts install.StartOpts) {
+	nodeIPPorts := make([]string, len(c.Nodes))
+	for i, node := range c.VMs {
+		if node.Provider == gce.ProviderName {
+			// only gce is supported for prometheus
+			desc, err := c.DiscoverService(ctx, install.Node(i+1), "", install.ServiceTypeUI, 0)
+			if err != nil {
+				l.Errorf("error getting the port for node %d: %v", i+1, err)
+				continue
+			}
+			nodeIPPorts[i] = fmt.Sprintf("%s:%d", node.PublicIP, desc.Port)
+		}
+	}
+	if len(nodeIPPorts) > 0 {
+		if err := promhelperclient.CreateClusterConfig(ctx,
+			envutil.EnvOrDefaultString(prometheusHostUrlEnv, defaultPrometheusHostUrl),
+			c.Name, nodeIPPorts); err != nil {
+			l.Errorf("creating cluster config failed: %v", err)
```

Comment:
Should we dump `nodeIPPorts` with the error message? In case, the cluster is long-running, it may still be possible to scrape it by manually editing the config. 

Also, methinks it would be useful to expose `createClusterConfig` behind a CLI flag, e.g., `roachprod prometheus $CLUSTER update-targets`. This way if the above call fails for any reason, it could be retried (on-demand) via CLI.